backend/models/upload.js

```diff
@@ -0,0 +1,240 @@
+	}
+};
+
+const handleChunkStream = async (req, filename) => {
+	return new Promise(resolve => {
+		const writeStream = fs.createWriteStream(filename, {encoding: "binary"});
+		req.on("data", (chunk) => writeStream.write(chunk, "binary"));
+		req.on("end", () => {
+			writeStream.end();
+			resolve();
+		});
+	});
+};
+
+const stitchChunks = (corID, newFilename) => {
+	const filePath = `${importQueue.getTaskPath(corID)}/chunks`;
+
+	const files = fs.readdirSync(filePath);
+
+	// sort timestamps into ascending order
+	files.sort((a, b) => a - b);
+
+	files.forEach((file) => {
+		// for debugging
+		const stats = fs.statSync(`${filePath}/${file}`);
+		systemLogger.logInfo(`FILE=${filePath}/${file}, SIZE=${stats.size} bytes`);
+		// end debugging
+
+		const content = fs.readFileSync(`${filePath}/${file}`);
+
+		fs.appendFileSync(`${importQueue.getTaskPath(corID)}/${newFilename}`, content);
+	});
+};
+
+const Upload = {};
+
+Upload.initChunking = async (teamspace, model, username, data) => {
+	if (!data.filename) {
+		throw responseCodes.INVALID_ARGUMENTS;
+	}
+
+	await checkFileFormat(data.filename);
+
+	// check model exists before upload
+	const modelSetting = await findModelSettingById(teamspace, model);
+
+	if (!modelSetting) {
+		throw responseCodes.MODEL_NOT_FOUND;
+	}
+
+	const newFileName = data.filename.replace(C.FILENAME_REGEXP, "_");
+
+	await isValidTag(teamspace, model, data.tag);
+
+	const corID = nodeuuid();
+
+	await importQueue.writeImportData(
+		corID,
+		teamspace,
+		model,
+		username,
+		newFileName,
+		data.tag,
+		data.desc,
+		data.importAnimations
+	);
+
+	return { corID };
+};
+
+Upload.uploadFile = async (req) => {
+	if (!config.cn_queue) {
+		return Promise.reject(responseCodes.QUEUE_NO_CONFIG);
+	}
+
+	const { account, model } = req.params;
+	const user = req.session.user.username;
+
+	modelStatusChanged(null, account, model, { status: "uploading", user });
+	// upload model with tag
+
+	const uploadedFile = await new Promise((resolve, reject) => {
+		const upload = multer({
+			dest: config.cn_queue.upload_dir,
+			fileFilter: async function(fileReq, file, cb) {
+				const size = parseInt(fileReq.headers["content-length"]);
+
+				try {
+					await checkFileFormat(file.originalname);
+					await middlewares.checkSufficientSpace(account, size);
+					cb(null, true);
+				} catch (err) {
+					cb(err);
+				}
+			}
+		});
+
+		upload.single("file")(req, null, function (err) {
+			if (err) {
+				return reject(err);
+
+			} else if(!req.file.size) {
+				return reject(responseCodes.FILE_FORMAT_NOT_SUPPORTED);
+
+			} else {
+				modelStatusChanged(null, account, model, { status: "uploaded" });
+				return resolve(req.file);
+			}
+		});
+	});
+
+	// req.body.tag wont be defined after the file has been uploaded
+	await isValidTag(account, model, req.body.tag);
+
+	return uploadedFile;
+};
+
+Upload.uploadChunksStart = async (teamspace, model, corID, username, headers) => {
+	if (!headers["x-ms-transfer-mode"] ||
```

Comment:
probably good to make these strings constants somewhere as you reference it more than once.